packages/hyperion-util/src/performanceAbsoluteNow.ts

```diff
@@ -38,15 +43,24 @@ if (performanceIsDefined) {
   if (performance.timing && performance.timing.navigationStart) {
     navigationStart = performance.timing.navigationStart;
   } else if (performance.timeOrigin) {
-    navigationStart = performance.timeOrigin
+    navigationStart = performance.timeOrigin;
   }
 }
 
+let coreFunction: GetTimeFunc;
 if (performanceIsDefined && typeof performance.now === "function" && navigationStart !== -1) {
-  performanceAbsoluteNow = () => performance.now() + navigationStart;
+  coreFunction = () => performance.now() + navigationStart;
 } else {
-  performanceAbsoluteNow = () => fallback();
+  coreFunction = () => fallback();
 }
-performanceAbsoluteNow.setFallback = setFallback;
+
+const extensions: GetTimeFuncExtensions = {
+  setFallback,
+  fromRelativeTime: (timestamp: number) => {
+    return timestamp + navigationStart;
```

Comment:
Could there be any unexpected offsets if `navigationStart` is still -1 at this point?

Reply:
I debated with myself on this one too. If `navigationStart` is still -1 at this point, then we will fallback to the Date.now and it means that the application also does not have access to `performance` and hence cannot pass relative values to the function. 
So, in practice, it should not happen. I add a warning to at least ensure this is not happening.